analysis/tools/InteractiveCLI/src/main/kotlin/de/maibornwolff/codecharta/tools/interactivecli/ParserDialogInterface.kt

```diff
@@ -0,0 +1,18 @@
+package de.maibornwolff.codecharta.tools.interactivecli
+
+import picocli.CommandLine
+
+interface ParserDialogInterface {
+    fun generateParserArgs(args: Array<String>, commandLine: CommandLine): Array<String>
+
+    fun isValidFileName(fileName: String): Boolean
+
+    fun getOutputFileName(fullFileName: String): String {
+        return fullFileName.substringAfterLast("/").substringAfterLast("\\").substringBefore(".")
+    }
+
+    fun checkExtension(fileName: String, expectedFileExtension: String): Boolean {
```

Comment:
rename to `hasMatchingExtension` or `hasExpectedExtension`. This implies that the function returns something AND that this something is a boolean. `checkExtension` sounds like a function, that does not return anything and throws errors in case, something does not match.

Reply:
I agree! Fixed in the next commits